syncapi/types/types.go

```diff
@@ -21,12 +21,38 @@ import (
 	"github.com/matrix-org/gomatrixserverlib"
 )
 
-// StreamPosition represents the offset in the sync stream a client is at.
-type StreamPosition int64
+// SyncPosition contains the PDU and EDU stream sync positions for a client.
+type SyncPosition struct {
+	// PDUPosition is the stream position for PDUs the client is at.
+	PDUPosition int64
+	// TypingPosition is the position for typing notifications the client is at.
+	TypingPosition int64
+}
 
 // String implements the Stringer interface.
-func (sp StreamPosition) String() string {
-	return strconv.FormatInt(int64(sp), 10)
+func (sp SyncPosition) String() string {
+	return strconv.FormatInt(sp.PDUPosition, 10) + "_" +
+		strconv.FormatInt(sp.TypingPosition, 10)
+}
+
+// IsAfter returns whether sp refers to states newer than other
```

Comment:
People calling this function won't know what `sp` or `other` are.

```suggestion
// IsAfter returns whether one SyncPosition refers to states newer than another SyncPosition.
```